corehq/apps/users/middleware.py

```diff
@@ -41,4 +39,7 @@ def process_view(self, request, view_func, view_args, view_kwargs):
                     request.couch_user = InvalidUser()
                 if request.couch_user:
                     request.couch_user.current_domain = domain
+        elif 'domain' in view_kwargs and '/reports/custom/' in request.path\
```

Comment:
I think this should be under `if request.user.is_anonymous() and 'domain' in view_kwargs:`